Dockerfile-taas

```diff
@@ -0,0 +1,52 @@
+FROM golang:1.19.5-alpine AS build
+LABEL maintainer="avinash@portworx.com"
+ARG MAKE_TARGET
+
+WORKDIR /go/src/github.com/portworx/torpedo
+
+# Install setup dependencies
+RUN apk update && apk add --no-cache bash git gcc musl-dev make curl openssh-client
+
+# No need to copy *everything*. This keeps the cache useful
+COPY vendor vendor
+COPY Makefile Makefile
+COPY go.mod go.mod
+COPY go.sum go.sum
+COPY pkg pkg
+COPY porx porx
+COPY scripts scripts
+COPY drivers drivers
+COPY deployments deployments
+COPY apiServer apiServer
+
+# Why? Errors if this is removed
+COPY .git .git
+
+# copy tests last to allow caching of the previous docker image layers
+COPY tests tests
+
+# Compile
+RUN --mount=type=cache,target=/root/.cache/go-build make $MAKE_TARGET
+
+# Build a fresh container with just the binaries
+FROM alpine
+
+RUN apk add --no-cache ca-certificates bash curl jq libc6-compat
+
+# clone the torpedo repo
+#RUN apk go git && git clone https://github.com/portworx/torpedo.git /taas
```

Comment:
This is ok for now. Let us create a binary for torpedo-api in the next PR and run it instead of cloning the whole repo